src/App.tsx

```diff
@@ -1,18 +1,136 @@
+/* eslint-disable react-hooks/exhaustive-deps */
 import { PeoplePage } from './components/PeoplePage';
 import { Navbar } from './components/Navbar';
 
 import './App.scss';
+import { Navigate, Route, Routes, useSearchParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { getPeople } from './api';
+import { Person } from './types';
+import { SortParams } from './components/types';
 
 export const App = () => {
+  const [loadingPeople, setLoadingPeople] = useState<boolean>(true);
+  const [error, setError] = useState<boolean>(false);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [visiblePeople, setVisiblePeople] = useState<Person[]>([]);
+
+  const [searchParams] = useSearchParams();
+
+  const filterSex = searchParams.get('sex') || '';
+  const filterQuery = searchParams.get('query') || '';
+  const filterCentury = searchParams.getAll('centuries') || [];
+  const filterSort = searchParams.get('sort') || '';
+  const filterOrder = searchParams.get('order') || '';
+
+  const getCentury = (year: number) => {
+    if (year < 0) {
+      return null;
+    }
+
+    return Math.ceil(year / 100);
+  };
+
+  //func to sort the array
+  const handleSortFunction = ({ peopleInFunc, sortType, desc }: SortParams) => {
+    const sortedPeople = [...peopleInFunc];
+
+    switch (sortType) {
+      case 'name':
+        sortedPeople.sort((a, b) => a.name.localeCompare(b.name));
+        break;
+      case 'sex':
+        sortedPeople.sort((a, b) => a.sex.localeCompare(b.sex));
+        break;
```

Comment:
```suggestion
      case 'name':
      case 'sex':
        sortedPeople.sort((a, b) => a[sort].localeCompare(b[sort]));
        break;
```
you can combine some cases